core/src/main/scala/org/apache/spark/deploy/master/ui/MasterPage.scala

```diff
@@ -80,15 +80,15 @@ private[ui] class MasterPage(parent: MasterWebUI) extends WebUIPage("") {
       .flatMap(_.toIterator)
       .groupBy(_._1) // group by resource name
       .map { case (rName, rInfoArr) =>
-        rName -> rInfoArr.map(_._2).reduce(_ + _)
-      }.map { case (k, v) => (k, v.toResourceInformation) }
+      rName -> rInfoArr.map(_._2.addresses.size).sum
+    }
     val usedInfo = aliveWorkers.map(_.resourcesInfoUsed)
       .map (resources => toMutable(resources))
```

Comment:
We can avoid this conversion now? (and same for line 79)

Reply:
good catch, yes we should be able to, let me update.